README.md

```diff
@@ -1,6 +1,190 @@
+
+To be able to run this tutorial:
+- [Sign In the CMP and created a workspace](https://www.stereolabs.com/docs/cloud/overview/get-started/).
+- [Add and Setup a device](https://www.stereolabs.com/docs/cloud/overview/get-started/#add-a-camera).
+- As most of the tutorials require a ZED camera, a ZED should be plugged to this device.
+
+## What is a CMP app ?
+A CMP app is an application deployed in a Docker container on one of the devices that you installed on your CMP workspace. Note that it can be coded in the langage of your choice and is not limited to a camera usage. It can be everything: a web server, a sensor reporter, an alert notifier, a node-red application ... However the CMP interface is optimized for the camera managment, and is especialy adapted to the ZED 3D cameras. Therefore most of the available tutorials requires a ZED to be runned.
+
+## Develop a CMP app
+
+In these [**CMP tutorials**](./tutorials/) you will learn how to develop a CMP app, exploring each feature.
+The only requirement is to setup a device to install the Edge Agent software and the sl_iot library.
+
+You can start it using this command, and stop it with CTRL+C :
+```
+$ edge_agent start
+```
+
+If you want to run it in backround use :
+```
+$ edge_agent start -b
+```
+
+And to stop it :
+```
+$ edge_agent stop
+```
+
+## Deploy a CMP app a service
+
+For production, you need to deploy your application on your device as a service, using Docker.
+Here is the full explanation on how to do it. Examples are available on [**Sample apps**](./samples/).
+
+### General structure
+A CMP app is runned in a Docker container. The [CMP documentation](https://www.stereolabs.com/docs/cloud/applications/) explains in detail how an app is structured and deployed. 
+To summury to deploy an app you need to upload a .zip file that contains at least:
+- docker-compose.yml describing how the application should be runned
+- a runtime dockerfile listing the steps and commands to compile and run the application.
+- an app.json file that descibes your application, specifies its name and release name and defines the parameters (see tutorial_04_application_parameters for more information about it)
+- the source code or the executable 
+
+#### Devlopping an app in C++
+You are adviced to developp your app in C++ insofar as a C++ API named sl_IOT is available in this langage and make it easier to use the CMP features.
+The provided tutorials can be a good starting point for your own apps. Here is how they should be used:
+
+```
+.
+├── app
+│   └── Dockerfile.runtime
+├── app.json
+├── cmp_builder.sh
+├── docker-compose.yml
+├── README.md
+└── sources
+    ├── CMakeLists.txt
+    ├── Dockerfile.build
+    └── src
+        └── main.cpp 
+```
+
+**Build stage**: 
+The source code needs to be compiled before deploying the app. The code is compiled inside a dedicated docker image. To run the associated container you just need to run 
+```
+$ ./cmp_builder.sh
```

Comment:
this will have to be updated with you current dev

Reply:
Done